docs/dev/development-setup.md

```diff
@@ -0,0 +1,85 @@
+# Development setup
+
+## Forking
+
+Please note that Go requires code to live under absolute paths, which complicates forking. While my fork lives at `https://github.com/colin-axner/ibc-go`, the code should never exist at `$GOPATH/src/github.com/colin-axner/ibc-go`. Instead, we use `git remote` to add the fork as a new remote for the original repo `$GOPATH/src/github.com/cosmos/ibc-go`), and do all the work there.
+
+For instance, to create a fork and work on a branch of it, I would:
+
+- Create the fork on GitHub, using the fork button.
+- Go to the original repo checked out locally (i.e. `$GOPATH/src/github.com/cosmos/ibc-go`)
+- `git remote add fork git@github.com:colin-axner/ibc-go.git`
+
+Now `fork` refers to my fork and `origin` refers to the ibc-go version. So I can `git push -u fork main` to update my fork, and make pull requests to ibc-go from there. Of course, replace `colin-axner` with your git handle.
+
+To pull in updates from the origin repo, run
+
+- `git fetch origin`
+- `git rebase origin/main` (or whatever branch you want)
+
+## Dependencies
+
+We use [Go 1.14 Modules](https://github.com/golang/go/wiki/Modules) to manage dependency versions.
+
+The main branch of every Cosmos repository should just build with `go get`, which means they should be kept up-to-date with their dependencies, so we can get away with telling  people they can just `go get` our software.
+
+Since some dependencies are not under our control, a third party may break our build, in which case we can fall back on `go mod tidy -v`.
+
+## Protobuf
+
+We use [Protocol Buffers](https://developers.google.com/protocol-buffers) along with [gogoproto](https://github.com/gogo/protobuf) to generate code for use in ibc-go.
+
+For determinstic behavior around protobuf tooling, everything is containerized using Docker. Make sure to have Docker installed on your machine, or head to [Docker's website](https://docs.docker.com/get-docker/) to install it.
+
+For formatting code in `.proto` files, you can run the `make proto-format` command.
+
+For linting and checking breaking changes, we use [buf](https://buf.build/). You can use the commands `make proto-lint` and `make proto-check-breaking` to respectively lint your proto files and check for breaking changes.
+
+To generate the protobuf stubs, you can run `make proto-gen`.
```

Comment:
We could add: to update third party protobuf dependencies, you can run `make proto-update-deps`.

This requires `buf` is installed in the local development environment.

```
brew install bufbuild/buf/buf

# OR

# Substitute PREFIX for your install prefix.
# Substitute VERSION for the current released version.
PREFIX="/usr/local" && \
VERSION="1.9.0" && \
curl -sSL \
    "https://github.com/bufbuild/buf/releases/download/v${VERSION}/buf-$(uname -s)-$(uname -m).tar.gz" | \
    tar -xvzf - -C "${PREFIX}" --strip-components 1
```

Reply:
Is there buf install documentation we can link to instead?